webapp/src/app.tsx

```diff
@@ -142,165 +52,19 @@ const App = (): JSX.Element => {
-                                            return <Redirect to={loginUrl}/>
-                                        } else if (loggedIn === true) {
-                                            if (continueToWelcomeScreen()) {
-                                                return <Redirect to={`/welcome?r=${originalPath}`}/>
-                                            }
-
-                                            return (
-                                                <BoardPage/>
-                                            )
-                                        }
-                                        return null
-                                    }}
-                                />
-                                <Route
-                                    exact={true}
-                                    path='/welcome'
-                                >
-                                    <WelcomePage/>
-                                </Route>
-
-                                {!Utils.isFocalboardPlugin() &&
-                                    <Route
-                                        path='/:boardId?/:viewId?/:cardId?'
-                                        render={({match: {params: {boardId, viewId, cardId}}}) => {
-                                        // Since these 3 path values are optional and they can be anything, we can pass /x/y/z and it will
-                                        // match this route however these values may not be valid so we should at the very least check
-                                        // board id for descisions made below
-                                            const boardIdIsValidUUIDV4 = UUID_REGEX.test(boardId || '')
-
-                                            if (loggedIn === false) {
-                                                return <Redirect to='/login'/>
-                                            }
-
-                                            if (continueToWelcomeScreen()) {
-                                                const originalPath = `/${Utils.buildOriginalPath('', boardId, viewId, cardId)}`
-                                                const queryString = boardIdIsValidUUIDV4 ? `r=${originalPath}` : ''
-                                                return <Redirect to={`/welcome?${queryString}`}/>
-                                            }
-
-                                            if (loggedIn === true) {
-                                                return <BoardPage/>
-                                            }
-
-                                            return null
-                                        }}
-                                    />}
-                            </Switch>
-                        </div>
+                <div id='frame'>
+                    <div id='main'>
+                        <NewVersionBanner/>
+                        <FocalboardRouter/>
```

Comment:
I removed everything here and replace it with the `FocalboardRouter` that is more or less the same but in an independent component.